### DISCUSSION TITLE
How to enable runtime logging on MacOS build

### DISCUSSION THREAD

**Discussion by ryanwwest** [koreader/koreader#10496] [Q&A]
I'm using a build of the macos koreader to do some experimenting and development. At the moment I'm not modifying any source code and was wondering how I could see the console log output when I interact with the app? I can do this via `./kodev run` and then see console output, but was wondering if this is the only option or if the log could be viewed when starting the app from the macos GUI icon.

**Comment by Frenzie** [comment]
I imagine you can run it from the console somehow, but if not you can always tail crash.log

**Reply by pazos** [reply]
On sdl targets stdout is not redirected to crash.log.

> how I could see the console log output when I interact with the app?

On macs you can use XCode instruments to run and profile the app. It should give you stdout and stderr too :)

You can also run the `koreader` binary from the app bundle. It is placed on  some "standard" path inside the application folder.
To get a `koreader` binary as in Linux you can do an alias or add the binary path to your $PATH

**Reply by johan456789** [reply]
> It is placed on some "standard" path inside the application folder.

I think you meant `/Applications/KOReader.app/Contents/MacOS`

Is it still the case that running from `KOReader.app` doesn't save the `crash.log` somewhere?
